tests/unit/stores/utilities/pool.spec.js

```diff
@@ -296,4 +272,377 @@ describe('pool', () => {
+        expect(tube.requests).toEqual([request.id])
+        expect(tube.source_id).toEqual(String(library.id))
+      })
+    })
+
+    it('create the used aliquots and map to source id', () => {
+      const storeLibraries = dataToObjectById({ data: libraries, includeRelationships: true })
+      const usedAliquots = createUsedAliquotsAndMapToSourceId({
+        libraries: storeLibraries,
+        aliquots,
+      })
+      aliquots.forEach((aliquot) => {
+        const usedAliquot = usedAliquots[`_${aliquot.attributes.source_id}`]
+        expect(usedAliquot.tag_id).toEqual(String(aliquot.attributes.tag_id))
+        // this is a bit painful. Once refactored this should be tested within usedAliquot.
+        // but it adds an extra check.
+        expect(usedAliquot.request).toEqual(storeLibraries[usedAliquot.source_id].request)
+        expect(usedAliquot.available_volume).toEqual(
+          parseFloat(
+            storeLibraries[usedAliquot.source_id].available_volume + usedAliquot.volume,
+          ).toFixed(2),
+        )
+      })
+    })
+  })
+
+  describe('in flight modifications for fetching tubes', () => {
+    const tubes = [
+      {
+        id: '20',
+        type: 'tubes',
+        attributes: {
+          barcode: 'TRAC-2-20',
+        },
+        relationships: {
+          pools: {
+            data: [],
+          },
+          libraries: {
+            data: {
+              type: 'libraries',
+              id: '30',
+            },
+          },
+        },
+      },
+    ]
+
+    const requests = [
+      {
+        id: '32',
+        type: 'requests',
+        attributes: {
+          library_type: 'Pacbio_HiFi',
+        },
+        relationships: {},
+      },
+      {
+        id: '33',
+        type: 'requests',
+        attributes: {
+          library_type: 'Pacbio_HiFi',
+        },
+        relationships: {},
+      },
+      {
+        id: '30',
+        type: 'requests',
+        attributes: {
+          library_type: 'Pacbio_HiFi',
+        },
+        relationships: {},
+      },
+    ]
+
+    const libraries = [
+      {
+        id: '30',
+        type: 'libraries',
+        attributes: {
+          volume: 1,
+          concentration: 1,
+          source_identifier: 'GEN-1710774222-1:F4',
+          pacbio_request_id: 30,
+          available_volume: 20,
+        },
+        relationships: {
+          request: {
+            data: {
+              type: 'requests',
+              id: '30',
+            },
+          },
+          tube: {
+            data: {
+              type: 'tubes',
+              id: '20',
+            },
+          },
+          primary_aliquot: {
+            data: {
+              type: 'aliquots',
+              id: '314',
+            },
+          },
+          used_aliquots: {
+            data: [
+              {
+                type: 'aliquots',
+                id: '315',
+              },
+            ],
+          },
+        },
+      },
+    ]
```

Comment:
Same question as above about data

Reply:
I think it needs to stay here because the factory uses those methods.

I have taken care to cut down the data as much as possible.

It also acts as a test. If we make any changes then the factory will break.